workflow-manager/src/domain/cosmos_cw.rs

```diff
@@ -652,6 +676,148 @@ impl Connector for CosmosCosmwasmConnector {
 
         Ok(())
     }
+
+    async fn query_workflow_registry(
+        &mut self,
+        main_domain: &str,
+        id: u64,
+    ) -> ConnectorResult<valence_workflow_registry_utils::WorkflowResponse> {
+        if self.chain_name != main_domain {
+            return Err(CosmosCosmwasmError::Error(anyhow::anyhow!(
+                "Adding external domain is only possible on main domain in authorization contract"
+            ))
+            .into());
+        }
```

Comment:
question: workflow registry will always be in Main domain right?

regarding error, shouldn't the error be different in this case as we are not adding external domain here (maybe copy pasted from a different place)